examples/wasi.rs

```diff
@@ -37,28 +37,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // Let's compile the Wasm module.
     let module = Module::new(&store, wasm_bytes)?;
 
-    println!("Creating `WasiEnv`...");
-    // First, we create the `WasiEnv`
-    let wasi_env = WasiState::new("hello")
+    // Run the module.
+    let mut wasi_env = WasiEnv::builder("hello")
         // .args(&["world"])
         // .env("KEY", "Value")
-        .finalize(&mut store)?;
```

Comment:
I think is still a good idea to separate the build of the state, and the initialize funciton.

I'm happy to have only two, but I don't think we should have one that does the two

Reply:
What could be the use of a seprate WasiState for the end user?

Reply:
Note that `WasiState` is currently a private implementation detail. 

To properly initialize a state we need the module and the FunctionEnv, because there are functional dependencies between them. 

The API is intended to prevent misuse, because it's rather easy to use a built state incorrectly.

Reply:
Still, it doesn't allow to set custom imports that may be provided by the user.
We need composability, as is also needed in wasmer-js and in the wai generated bindings for WASI modules.

In general, we should have something like:

```rust
    let import_object = wasi_env.import_object(&mut store, &module)?;
    let instance = Instance::new(&mut store, &module, &import_object)?;

   wasi_env.initialize(instance);
```

Reply:
Still possible - a new example is at `examples/wasi_manual_setup.rs`.